cmd/mt-store-cat/main.go

```diff
@@ -29,43 +29,41 @@ var (
+	flag.StringVar(&storeConfig.HostSelectionPolicy, "cassandra-host-selection-policy", storeConfig.HostSelectionPolicy, "")
+	flag.IntVar(&storeConfig.Timeout, "cassandra-timeout", storeConfig.Timeout, "cassandra timeout in milliseconds")
+	flag.IntVar(&storeConfig.ReadConcurrency, "cassandra-read-concurrency", storeConfig.ReadConcurrency, "max number of concurrent reads to cassandra.")
+	//flag.IntVar(&storeConfig.WriteConcurrency, "write-concurrency", storeConfig.WriteConcurrency, "max number of concurrent writes to cassandra.")
+	flag.IntVar(&storeConfig.ReadQueueSize, "cassandra-read-queue-size", storeConfig.ReadQueueSize, "max number of outstanding reads before reads will be dropped. This is important if you run queries that result in many reads in parallel.")
+	//flag.IntVar(&storeConfig.WriteQueueSize, "write-queue-size", storeConfig.WriteQueueSize, "write queue size per cassandra worker. should be large engough to hold all at least the total number of series expected, divided by how many workers you have")
+	flag.IntVar(&storeConfig.Retries, "cassandra-retries", storeConfig.Retries, "how many times to retry a query before failing it")
+	flag.IntVar(&storeConfig.WindowFactor, "window-factor", storeConfig.WindowFactor, "size of compaction window relative to TTL")
+	flag.IntVar(&storeConfig.OmitReadTimeout, "cassandra-omit-read-timeout", storeConfig.OmitReadTimeout, "if a read is older than this, it will directly be omitted without executing")
+	flag.IntVar(&storeConfig.CqlProtocolVersion, "cql-protocol-version", storeConfig.CqlProtocolVersion, "cql protocol version to use")
+	flag.BoolVar(&storeConfig.CreateKeyspace, "cassandra-create-keyspace", storeConfig.CreateKeyspace, "enable the creation of the mdata keyspace and tables, only one node needs this")
+	flag.BoolVar(&storeConfig.DisableInitialHostLookup, "cassandra-disable-initial-host-lookup", storeConfig.DisableInitialHostLookup, "instruct the driver to not attempt to get host info from the system.peers table")
+	flag.BoolVar(&storeConfig.SSL, "cassandra-ssl", storeConfig.SSL, "enable SSL connection to cassandra")
+	flag.StringVar(&storeConfig.CaPath, "cassandra-ca-path", storeConfig.CaPath, "cassandra CA certificate path when using SSL")
+	flag.BoolVar(&storeConfig.HostVerification, "cassandra-host-verification", storeConfig.HostVerification, "host (hostname and server cert) verification when using SSL")
+	flag.BoolVar(&storeConfig.Auth, "cassandra-auth", storeConfig.Auth, "enable cassandra authentication")
+	flag.StringVar(&storeConfig.Username, "cassandra-username", storeConfig.Username, "username for authentication")
+	flag.StringVar(&storeConfig.Password, "cassandra-password", storeConfig.Password, "password for authentication")
+	flag.StringVar(&storeConfig.SchemaFile, "cassandra-schema-file", storeConfig.SchemaFile, "File containing the needed schemas in case database needs initializing")
```

Comment:
yeah, i think it would be nice if the `storeConfig` had a method that accepts a flag set as argument and then just adds all of those cassandra settings to the given flag set. wouldn't that save a lot of redundant flag definitions?

Reply:
If you have more refactoring planned, can you add it to this PR?
i notice some discrepancies in how tools define flags, it's time consuming to go over them, and if you want to redo it anyway, let's save time and do it in one shot.

Reply:
interesting. flagsets (sections) in globalconf are different than what flag sets in the standard library are used for.
we need to configure multiple independent modules, and created multiple sections/flagsets accordingly. i was thinking maybe we could specify the flags by including the section name in the flags (similar to how env vars work in globalconf) but that's not how it is supposed to work when parsing flags.
flagsets are actually meant to provide configuration pertaining to exactly 1 "action" (subcommand) which needs to be specified on the command line.
these 2 links show how flagsets are meant to be used.
* https://groups.google.com/forum/#!topic/golang-nuts/1lSH5rY5IhQ
* https://blog.rapid7.com/2016/08/04/build-a-simple-cli-tool-with-golang/ (go to "subcommands")

Reply:
> If you have more refactoring planned, can you add it to this PR?

no, it is way outside the scope of this PR.